lib/os.ml

```diff
@@ -94,26 +94,22 @@ let write_file ~path contents =
   Lwt_io.write ch contents
 
 let with_pipe_from_child fn =
-  let r, w = Lwt_unix.pipe_in () in
+  let r, w = Lwt_unix.pipe_in ~cloexec:true () in
+  Unix.clear_close_on_exec w;
```

Comment:
I think we should keep both FDs set to close-on-exec here (the original version was wrong too). If we fork another process after this point and before closing `w` then `w` will leak to it. We should only clear close-on-exec right before we do the exec.